src/test/java/cz/inventi/kpj/kpjtesting/KpjTestingTest.java

```diff
@@ -10,41 +19,40 @@
  * 4. Change behavior of {@link HelloWorldServiceImpl#echo(String)} to always
  * returns "KPJ Rulez" <br/>
  */
+@SpringBootTest
 @MockitoSettings(strictness = Strictness.LENIENT)
 public class KpjTestingTest {
-
-
+  @Spy
   HelloWorldService helloWorldService = new HelloWorldServiceImpl();
 
-
+  @InjectMocks
   KpjTesting kpjTesting;
 
+  @BeforeEach
   void setUp() {
-    // setup mocks for all tests
-    // hint you have to use doReturn for Spy
     // Change behavior of HelloWorldServiceImpl#echo(String)
-    // to always returns "KPJ Rulez"
-
+    // to always return "KPJ Rulez"
+    doReturn("KPJ Rulez").when(helloWorldService.echo("Print ECHO"));
```

Comment:
It is wrong syntax. You can use calling method inside a method only for `Mockito.when()`.
`Mockito.verify()` and `Mockito.doXXX.when()` these methods return T and you call the method on T. 
e.g.:
```suggestion
    doReturn("KPJ Rulez").when(helloWorldService).echo("Print ECHO");
```